Update branch policy (#5363)
plone/volto: docs/source/contributing/branch-policy.md

    When opening a pull request, the contributor must open it against `main`.
    If the pull request is a feature or a bugfix, and if the release manager deems it useful to the latest version's branch, they may ask the contributor to backport it to that branch.

`17.x.x`
:   `17.x.x` is the current stable and actively developed branch in git.
    Upon the final release of version `18.0.0`, the `17.x.x` branch line will be no longer supported.

>     Any new feature will be generally released to the `main` branch, and only backported to older versions if the Volto Team approves it.

Reviewer: @sneridagh The word "released" is confusing here. You mean "merged", right?

Reviewer 2: How about this?

```
    Any new feature will be merged into the `main` branch, and only backported to and released in older versions, if the Volto Team approves it.
```

    When opening a pull request, the contributor must open it against `main`.
    If the pull request is a feature or a bugfix, and if the release manager deems it useful to the latest version's branch, they may ask the contributor to backport it to that branch.

`17.x.x`
:   At the moment of this writing, `17.x.x` is the current stable branch in git.
    Upon the final release of version `18.0.0`, the `17.x.x` branch line will become legacy.
    This is the current actively developed branch in git, meaning that it may receive new features and bug fixes.

>     Any new feature will be released generally to the `main` branch, and only backported if the Volto Team approves it.

Reviewer 2: I'd simplify a bit. I'm also concerned about the loose use of "legacy" throughout this policy. I think we need to explicitly define "legacy" as a term for Volto and all of Plone.

```
:   `17.x.x` is the current stable and actively developed branch in git.
    It may receive new features and bug fixes.
    Upon the final release of version `18.0.0`, the `17.x.x` branch line will become legacy.
    Any new feature will be generally released to the `main` branch, and only backported to older versions if the Volto Team approves it.
```

Author: It's the term we decided during the Volto Team Meeting at the conf. Don't know if we have to rethink it, but let's discuss it. I think @davisagli gave his ok during the meeting.
No, we decided that the "legacy" versions won't receive anything, unless it's justified and approved by the Team. 
This is to reduce the overall burden on the Release Manager (eg. when we have five legacy versions).

Author: jeez I just remembered you couldn't attend :( It would have been great to have you there, although few things, they were important. It's only now that I had time to write them down (and I'm cutting the 17.x.x branch now too).

Reviewer 2: Why do we need YAOT (yet another obfuscating term)?

Let's just call it "unsupported" or "no longer supported" (which would be consistent with the term already used in the [Plone Release Schedule](https://plone.org/download/release-schedule)), and drop "legacy".

Author: +1 good enough for me

/cc @plone/volto-team @davisagli thoughts? If any, we will go with it.

Reviewer 2: Updated suggestion, replacing "legacy" with "no longer supported".

Author: Done.

`16.x.x`

> :   This is the LTS version of Volto, aligned with Plone 6.0 release.

Reviewer 2: See https://github.com/plone/volto/pull/4905 and https://github.com/plone/volto/issues/4904

Author: We also talked about this one... although is controversial, we decided to stick with the denomination not as a LTS per se, but to align with the Plone LTS (6.0).

Reviewer 2: I think it would be clearer to say this, and avoid "LTS".

```suggestion
:   This version of Volto has the [same maintenance and support schedule as Plone 6.0](https://plone.org/download/release-schedule).
```

    It receives bug fixes and security updates.

>     Any new feature will be released generally to the `main` branch, and only backported if the Volto Team approves it.

Reviewer 2: ```suggestion
    Any new feature will be released generally to the `main` branch, and only backported to `16.x.x` if the Volto Team approves it.
```

`15.x.x`

> :   All branches below 15.x.x (14.x.x, 13.x.x, and so on) are considered legacy branches and won't receive any update.

Reviewer 2: ```suggestion
:   `15.x.x` and older branches are considered legacy branches and won't receive any updates.
```

>     If you need a bug fix or security update to any of them, please send a PR and the Volto Team will review if it's suitable for merging.

Reviewer 2: ```suggestion
    If you need a bug fix or security update to any of them, please submit a pull request, and the Volto Team will review it to determine whether it's suitable to merge.
```

    Any new feature will be released generally to the `main` branch, and only backported if the Volto Team approves it.

`16.x.x`
:   This version of Volto has the [same maintenance and support schedule as Plone 6.0](https://plone.org/download/release-schedule).
    It receives bug fixes and security updates.
    Any new feature will be released generally to the `main` branch, and only backported to `16.x.x` if the Volto Team approves it.

`15.x.x`

> :   `15.x.x` and older branches are considered legacy branches and won't receive any updates.

Reviewer 2: ```suggestion
:   `15.x.x` and older branches are no longer supported.
```
